cli/src/rich/mod.rs

```diff
@@ -358,13 +531,238 @@ fn draw_input(screen: &mut dyn Write, input: &Input) -> io::Result<()> {
         termion::cursor::Goto(4 + input.cursor as u16, input_row)
     )?;
 
+    if let Some(suggestion) = suggestion.map(|it| it.term.as_ref()) {
+        write!(
+            screen,
+            "{}{}{}{}{}{}{}",
+            termion::cursor::Save,
+            termion::color::Fg(termion::color::LightWhite),
+            termion::color::Bg(termion::color::Blue),
+            suggestion
+                .chars()
+                .skip(input.get_text().len())
+                .collect::<String>(),
+            termion::color::Reset.fg_str(),
+            termion::color::Reset.bg_str(),
+            termion::cursor::Restore,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn draw_autocomplete(
+    screen: &mut dyn Write,
+    autocomplete: Option<&Autocomplete>,
+) -> io::Result<()> {
+    let Some(autocomplete) = autocomplete else {
+        return Ok(());
+    };
+
+    let Some(term_width) = autocomplete.suggestions.iter().map(|i| i.term.len()).max() else {
+        return Ok(());
+    };
+    let Some(summary_width) = autocomplete
+        .suggestions
+        .iter()
+        .map(|i| i.summary.len())
+        .max()
+    else {
+        return Ok(());
+    };
+    let width = term_width + summary_width + 2;
+
+    let (_, term_height) = termion::terminal_size().unwrap();
+    let start_row = term_height - 2 - autocomplete.len() as u16;
+
+    for (pos, suggestion) in autocomplete.suggestions.iter().enumerate() {
+        let padding =
+            String::from_utf8(vec![
+                b' ';
+                width - suggestion.term.len() - suggestion.summary.len()
+            ])
+            .unwrap();
+        let query_len = autocomplete.query.len();
+        let line = pos as u16 + start_row;
+
+        // Go to start position, draw opening border.
+        write!(
+            screen,
+            "{}{}{} ",
+            termion::cursor::Goto(3, line),
+            termion::color::Fg(termion::color::White),
+            termion::color::Bg(termion::color::LightBlack),
+        )?;
+
+        // This is a little indimidating but goes like so:
+        // - Switch to the 'match color' and draw the matching text
+        // - Switch to the text color, draw unmatched text
+        // - Switch to summary color, draw padding and the summary.
+        if Some(pos) == autocomplete.selected {
```

Comment:
I'm not a huge fan of the duplication here, but the way termion creates these colors ends up with totally different structs that all implement a `Color` trait.

Started down the path of defining an `AutocompletePalette` structure to capture these but it ended up being more code that felt more difficult to read than this.